File: app/src/main/java/player/phonograph/views/basic/CheckBoxX.java
package player.phonograph.views.basic;

import android.content.Context;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatCheckBox;

import util.mdcolor.pref.ThemeColor;
import util.mddesign.core.Themer;


/**
 * @author Aidan Follestad (afollestad)
 */
public class CheckBoxX extends AppCompatCheckBox {

    public CheckBoxX(Context context) {
        super(context);
        init(context, null);
    }

    public CheckBoxX(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    public CheckBoxX(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        Themer.setTint(this, ThemeColor.accentColor(context));
    }
}

File: app/src/main/java/lib/phonograph/view/CheckBoxX.kt
/*
 * Copyright (c) 2022 Abou Zeid (kabouzeid) (original author)
 */
package lib.phonograph.view

import mt.pref.ThemeColor.accentColor
import mt.tint.viewtint.setTint
import player.phonograph.util.theme.nightMode
import androidx.appcompat.widget.AppCompatCheckBox
import android.content.Context
import android.util.AttributeSet

/**
 * @author Aidan Follestad (afollestad)
 */
class CheckBoxX : AppCompatCheckBox {
    constructor(context: Context) : super(context) {
        init(context, null)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init(context, attrs)
    }

    private fun init(context: Context, attrs: AttributeSet?) {
        this.setTint(accentColor(context), context.nightMode)
    }
}